src/main/java/com/example/bookstorewebapp/repository/impl/BookRepositoryImpl.java

```diff
@@ -52,4 +53,15 @@ public List<Book> findAll() {
             throw new RuntimeException("Can't get all books", e);
         }
     }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            Book book = session.find(Book.class, id);
+            return Optional.ofNullable(book);
+
+        } catch (Exception e) {
+            throw new RuntimeException("Can't find book with id: " + id, e);
```

Comment:
```suggestion
            throw new EntityNotFoundException("Can't find book with id: " + id, e);
```

Reply:
Actually, I wouldn't agree here with @andrii-hoienko, I do not think this exception should be used in repository layer, because we can get basically any exception here, like SQL syntax problem or a connection to db. and that does not always mean that entity does not exist. 
So I'd recommend reverting this change and the one similar to it below